sale_stock_picking_note/models/sale_stock.py

```diff
@@ -1,15 +1,26 @@
 # Copyright 2018 Tecnativa - Carlos Dauden
 # Copyright 2021 Tecnativa - Víctor Martínez
+# Copyright 2023 Moduon Team S.L.
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    picking_note = fields.Html(string="Picking Internal Note")
-    picking_customer_note = fields.Text(string="Picking Customer Comments")
+    picking_note = fields.Html(
+        string="Picking Internal Note",
+    )
+    picking_customer_note = fields.Text(
+        string="Picking Customer Comments",
+    )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id_picking_notes(self):
+        for order in self:
+            order.picking_note = order.partner_id.picking_note
```

Comment:
question: what happens if there are user default values? Maybe we could avoid writing the notes if the user doesn't have any.

Reply:
I don't undestand you. User default values? Can you give me more context please?

Reply:
I think he means an `ir.default`. IMHO the partner defaults should override user defaults.

Reply:
@EmilioPascual imagine you don't have this improvement and you have a default vaule for sale order with this customer.

@chienandalu Default vaules has prioriti over the partner fields so I guess this will not impact to you, but please! tell us.

I have record a 3 minutes video! (check @EmilioPascual )

https://www.loom.com/share/4fa5b85fa6084a468fc65da91355f841?sid=ddccdcfa-54a1-4d84-94ee-215cb40859c6

Reply:
> IMHO the partner defaults should override user defaults.

IMHO the oposite 😄, default in sales should overwrite partner notes and in fact actually functionality I think is RIGHT! 👍🏼 as expected. Take all care that ir.default will be never used with this improvement. Further more I said to Emilio to open a new module consciously of this problems impacting in a new one, we don't want or have time to discuss too much this trivial improvement 😄 https://github.com/OCA/sale-workflow/pull/2830 (that why it was other module)

Reply:
> > IMHO the partner defaults should override user defaults.
> 
> IMHO the oposite 😄, default in sales should overwrite partner notes and in fact actually functionality I think is RIGHT! 👍🏼 as expected.

🤔 why?

A salesman can sell to 1000 customers, where 900 have no specific note, but 100 of them have. Doesn't it make more sense if those 100 customers get their specific notes, instead of getting the default one that the salesman just uses for everyone else?